sdk/core/core/src/az_http_response.c

```diff
@@ -51,7 +51,7 @@ static AZ_NODISCARD az_result _az_get_digit(az_span* self, uint8_t* save_here)
   *save_here = (uint8_t)(c_ptr - '0');
 
   // move reader after the expected digit (means it was parsed as expected)
-  *self = az_span_slice(*self, 1, -1);
+  *self = az_span_slice_start(*self, 1);
```

Comment:
yeah, keeps reading weird..  we use the `from` anywhere
AZ_SPAN_FROM_BUFFER
AZ_SPAN_FROM_STR

why it does not apply here?